src/scripts/components/joinForm.js

```diff
@@ -0,0 +1,116 @@
+import { entries as getFormEntries } from '../utilities/formData';
+
+class JoinForm {
+
+  constructor( $element) {
+
+    this.$form = $($element);
+
+    this.$name = this.$form.find('[name="name"]');
+    this.$zip = this.$form.find('[name="zip"]')
+    this.$email = this.$form.find('[name="email"]');
+    this.$mobile = this.$form.find('[name="mobile_phone"]');
+
+    this.$page = this.$form.find('[name="page"]');
+    this.$posted = this.$form.find('[name="posted"]');
+    this.$formName = this.$form.find('[name="form_name"]');
+    this.$source = this.$form.find('[name="source"]');
+    this.$actionReferrer = this.$form.find('[name="action_referrer"]');
+    this.$country = this.$form.find('[name="country"]');
+
+    this.$subscribeSMSConditions = this.$form.find('.subscribe-sms-conditions');
+    this.$suppressEmailSubscribe = this.$form.find('[name="suppress_subscribe"]');
+    this.$smsTerms = this.$form.find('[name="user_sms_termsandconditions"]');
+    this.$smsSubscribe = this.$form.find('[name="action_mobilesubscribe"]');
+    this.$robodialTerms = this.$form.find('[name="user_robodial_termsandconditions"]');
+    this.$userSmsSubscribe = this.$form.find('[name="user_sms_subscribed"]');
+
+    this.$submit = this.$form.find('[type="submit"]');
+
+    this.$errors = this.$form.find('.errorlist');
+
+    this.formData = {};
+    this.bindform();
+  }
+
+  bindform() {
+    let _this = this;
+
+    this.$form.on('change keyup', function () {
+      if (_this.$mobile[0].value.replace(/\D/g,'').length >= 10) {
+        _this.$userSmsSubscribe[0].checked = true;
+        if (_this.$subscribeSMSConditions[0].className.search("unhide") == -1) {
+          _this.$subscribeSMSConditions[0].className += " unhide";
+        }
+      }
+    });
+
+    this.$form.on('submit', function (e) {
+      const Mobile = _this.$mobile[0];
+      const Email = _this.$email[0];
+      const Name = _this.$name[0];
+      const Zip = _this.$zip[0];
+      const fields = {Mobile, Email, Name, Zip};
+
+      _this.$errors[0].innerHTML = '';
+      Object.keys(fields).forEach(key =>
+        fields[key].parentNode.classList.remove("giraffe-has-errors")
+      );
+
+      if (Email.value.replace(/\D/g, '') === '' && Mobile.value === '' || Name.value === '' || Zip.value === '') {
+        e.preventDefault();
+        Object.keys(fields).forEach(function(key) { 
+          if (fields[key].value === '') {
+            fields[key].parentNode.classList.add("giraffe-has-errors");
+            if (key !== "Email") {
+              _this.addErrorToList(key);
+            }
+          }
+        });
+        return !1;
```

Comment:
is this meant to be a minimized `false`?  something else is minimizing this anyway, right?  Let's just say `false`